### PR TITLE
Plan: [Story] Automatic Baseline (Day Zero) for Score History #381

### LINKED ISSUE #381: [Story] Automatic Baseline (Day Zero) for Score History
## Objective
Ensure that tournament performance charts always start from a zero-point baseline by automatically creating a "Day Zero" entry the first time a user's score history is recorded.

## Problem / Background
Currently, when a scoring event occurs, a snapshot is saved in tournament_score_history. However, if this is the first entry for the user, their chart starts directly at their current score. This makes it look like they jumped from "nothing" to their first-day score. By recording a snapshot for the day before the first score, we provide a consistent origin (0 points) for all progression charts.

## Acceptance Criteria
- [ ] writeScoreSnapshot checks if a user has any existing history entries for the given tournament.
- [ ] If no history exists, it first creates an initial entry for the day before the requested snapshot_date.
- [ ] This "Day Zero" entry has 0 points in all six score segments (total_game_score, total_boost_bonus, etc.).
- [ ] Date calculation for "Day Zero" correctly handles month/year boundaries (e.g., Jan 1st → Dec 31st).
- [ ] The original requested snapshot is then written as usual.
- [ ] Subsequent scoring events for the same user/tournament do not trigger another "Day Zero" creation.

## Out of Scope
- Retroactive creation of Day Zero entries for users who already have score history data.
- Applying this baseline logic to the group_rankings table (this story focuses on individual tournament score history).

## Open Questions
- None.

### PR DESCRIPTION
**Pull request by gvinokur**
Fixes #381

## Summary
Implementation plan for the story.

## Plan Document
See `plans/STORY-381-plan.md` for full details.

## Next Steps
- Review and approve plan
- Iterate on plan based on feedback
- Execute plan once approved

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plans/STORY-381-plan.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+| `__tests__/utils/date-utils.test.ts` | Modify — add tests for `getPreviousDayYYYYMMDD` |
+| `__tests__/db/score-history-repository.test.ts` | Modify — add tests for `hasScoreHistory` and Day Zero behavior |
+| `docs/code-structure/utils.md` | Modify — document new function |
+| `docs/code-structure/db.md` | Modify — update `writeScoreSnapshot`, document `hasScoreHistory` |
+
+## Mid-Level Design
+
+### Call Graph Changes
+
+No call graph changes. `writeScoreSnapshot` gains an internal dependency on `hasScoreHistory` (same module) and `getPreviousDayYYYYMMDD` (utils), but no new cross-layer call relationships are introduced. The function signature is unchanged and all callers remain unmodified.
```
Comment (gvinokur) [inline plans/STORY-381-plan.md:81]: Except getPreviousDayYYYYMMDD is in utils, isn’t this technically a new call?
